fault-proof/src/challenger.rs

```diff
@@ -125,11 +129,29 @@ where
         };
 
         let Some(latest_index) = self.factory.fetch_latest_game_index().await? else {
+            tracing::warn!("No games to fetch (factory returned None)");
```

Comment:
I believe we don't need to warn this, it's possible scenario.
```suggestion
            tracing::debug!("No games to fetch (factory returned None)");
```